calendar/engine/availability.go

```diff
@@ -660,27 +645,32 @@ func (m *mscalendar) notifyUpcomingEvents(mattermostUserID string, events []*rem
 	}
 }
 
-func filterBusyEvents(events []*remote.Event) []*remote.Event {
+func filterBusyAndAttendeeEvents(events []*remote.Event) []*remote.Event {
 	result := []*remote.Event{}
 	for _, e := range events {
-		if e.ShowAs == "busy" && !e.IsCancelled {
+		// Not setting custom status for events without attendees since those are unlikely to be meetings.
+		if e.ShowAs == "busy" && !e.IsCancelled && len(e.Attendees) >= 1 {
 			result = append(result, e)
 		}
 	}
 	return result
 }
 
-// Function to check for overlapping events.
-func checkOverlappingEvents(events []*remote.Event) bool {
+// getMergedEvents accepts a sorted array of events, and returns events after merging them, if overlapping or if the meeting duration is less than StatusSyncJobInterval.
+func getMergedEvents(events []*remote.Event) []*remote.Event {
 	if len(events) <= 1 {
-		return false
+		return events
 	}
 
+	idx := 0
 	for i := 1; i < len(events); i++ {
-		if events[i-1].End.Time().UnixMicro() > events[i].Start.Time().UnixMicro() {
-			return true
+		if (events[idx].End.Time().UnixMicro() >= events[i].Start.Time().UnixMicro()) || (events[idx].End.Time().Sub(events[idx].Start.Time()) <= StatusSyncJobInterval && events[i].Start.Time().Sub(events[idx].End.Time()) <= StatusSyncJobInterval) {
+			events[idx].End = events[i].End
+		} else {
+			idx++
+			events[idx] = events[i]
 		}
 	}
 
-	return false
+	return events[0 : idx+1]
```

Comment:
It would be good to put the body of this loop in its own function, and have it unit tested to show/document how it works

Reply:
@mickmister Added a comment to it. I don't think we need to separately test the function as it is already covered by its parent function.